CMakeLists.txt

```diff
@@ -1,11 +1,36 @@
 # CMake4GDAL project is distributed under MIT license. See accompanying file LICENSE.txt.
+# Modified by Mi Zhang, LuoJiaNET Development and Research Group, Wuhan University, 2022
+
 cmake_minimum_required(VERSION 3.9...3.23)
 
 project(gdal LANGUAGES C CXX)
 include(CTest)
 
 set(GDAL_LIB_TARGET_NAME GDAL)
 
+###############################################################################
+# Generate setup.cfg
+###############################################################################
+if (WIN32)
+    set(GDAL_COMPILER mingw32)
+elseif(UNIX OR APPLE)
+    set(GDAL_COMPILER unix)
+else ()
+    message(FATAL_ERROR
+            "Python build_ext compiler inference is only supported on Win, Unix or Apple.")
+endif()
+message("Python build_ext compiler is inferred as '${GDAL_COMPILER}'.")
+
+if(NOT EXISTS ${CMAKE_SOURCE_DIR}/swig/python/setup.cfg)
+	configure_file(${CMAKE_SOURCE_DIR}/swig/python/setup.cfg.in
```

Comment:
> > in the source directory, instead, it generates the setup.cfg from setup.cfg.in.
> 
> see my later comment. We can't do that for now since we support autoconf/nmake build systems as well, for which swig/python/setup.cfg must be there / in the repository.

Thanks for the reply! We've tested it in many computers with win-10, mingw command line environment. Hope this branch can helpful to those with MinGW compiler.